### PR TITLE
add doc on provider delivery control

### PR DESCRIPTION
**Pull request by mmulholla**
Added documentation on provider delivery control. Also added some tables of content to make it easier to find information.

### CODE REVIEW

**Review by isutton — APPROVED** [review]

File: docs/helm-chart-checks.md [hunk 1]
```diff
@@ -13,6 +30,7 @@ Helm chart checks are a set of checks against which the Red Hat Helm chart-verif
     - Profiles are versioned. Each new version may include updated checks, new checks, new annotations or chnaged annotations.
```
Comment (Srivaralakshmi) [inline docs/helm-chart-checks.md:30]: Profiles are versioned. Each new version can include updated checks, new checks, new annotations, or changed annotations.

File: docs/helm-chart-checks.md [hunk 2]
```diff
@@ -13,6 +30,7 @@ Helm chart checks are a set of checks against which the Red Hat Helm chart-verif
...
 - The generated report is written to stdout but can optionally be written to a file.
```
Comment (Srivaralakshmi) [inline docs/helm-chart-checks.md:31]: L31: Rewrite for clarity.

File: docs/helm-chart-checks.md [hunk 3]
```diff
@@ -13,6 +30,7 @@ Helm chart checks are a set of checks against which the Red Hat Helm chart-verif
...
 - An error log is created for all verify commands but can be optionally suppressed.
+- You can indicate that a chart is not to be published in the OpenShift catalogue
```
Comment (Srivaralakshmi) [inline docs/helm-chart-checks.md:33]: You can indicate that a chart is not to be published in the OpenShift catalog.

File: docs/helm-chart-checks.md [hunk 4]
```diff
@@ -324,7 +343,7 @@ This table shows which checks are preformed and whether or not they ar mnandator
 | [chart-testing v1.0](helm-chart-troubleshooting.md#chart-testing-v10) | mandatory | mandatory | optional | mandatory
 | [contains-values v1.0](helm-chart-troubleshooting.md#contains-values-v10)  | mandatory | mandatory | optional | mandatory
 
-#### Running the chart verifier with a specific profile.
+### Running the chart verifier with a specific profile.
```
Comment (Srivaralakshmi) [inline docs/helm-chart-checks.md:346]: Remove the period from the end of the section title.

File: docs/helm-chart-submission.md [hunk 5]
```diff
@@ -16,7 +21,39 @@ The following options are available for submitting a chart for Red Hat OpenShift
...
+
 For more information on the submission process, see: [OpenShift Helm Charts Repository documentation](https://github.com/openshift-helm-charts/charts/blob/main/docs/README.md).
 
 For troubleshooting report related submission failures see: [Troubleshooting](./helm-chart-troubleshooting.md)
- 
\ No newline at end of file
+
+## Provider controlled delivery
+
+By default a submitted chart will be made available in the OpenShift Helm Chart Catalogue on successful certification. In some cases this is undesirable and can be prevented based on two conditions:
```
Comment (Srivaralakshmi) [inline docs/helm-chart-submission.md:32]: By default, a submitted chart will be made available in the OpenShift Helm Chart Catalog on successful certification. In some cases this is undesirable and can be prevented based on the following conditions:

File: docs/helm-chart-submission.md [hunk 6]
```diff
@@ -16,7 +21,39 @@ The following options are available for submitting a chart for Red Hat OpenShift
...
+
+1. When generating the Verification report the ```--provider-delivery``` flag is used, for example:
```
Comment (Srivaralakshmi) [inline docs/helm-chart-submission.md:34]: When generating the Verification report the ```--provider-delivery``` flag is used.
**Example:**

File: docs/helm-chart-submission.md [hunk 7]
```diff
@@ -16,7 +21,39 @@ The following options are available for submitting a chart for Red Hat OpenShift
...
+    ```
+    $ podman run --rm -i                                  \
+          -e KUBECONFIG=/.kube/config                   \
+          -v "${HOME}/.kube":/.kube                     \
+          "quay.io/redhat-certification/chart-verifier" \
+          verify --provider-delivery                    \
+          <chart-uri>
+    ```
+    This ensures the [providerControlledDelivery annotation](helm-chart-annotations.md#providerControlledDelivery) is set to True in the verification report.
```
Comment (Srivaralakshmi) [inline docs/helm-chart-submission.md:43]: This ensures that the [`providerControlledDelivery` annotation](helm-chart-annotations.md#providerControlledDelivery) is set to the value `True` in the verification report.

File: docs/helm-chart-submission.md [hunk 8]
```diff
@@ -16,7 +21,39 @@ The following options are available for submitting a chart for Red Hat OpenShift
...
+
+1. The OWNERS file for the submitted chart in the [openshift helm charts github repository](https://github.com/openshift-helm-charts/charts) includes a ```providerDelivery``` attribute which is set to True, for example:
```
Comment (Srivaralakshmi) [inline docs/helm-chart-submission.md:45]: check the list numbering. Rendering seems ok though.

Comment (Srivaralakshmi) [inline docs/helm-chart-submission.md:45]: The OWNERS file for the submitted chart in the [OpenShift Helm Charts GitHub repository] (https://github.com/openshift-helm-charts/charts) includes a `providerDelivery` attribute that is set to the value `True`.
**Example:**

File: docs/helm-chart-submission.md [hunk 9]
```diff
@@ -16,7 +21,39 @@ The following options are available for submitting a chart for Red Hat OpenShift
...
+publicPgpKey: null
+providerDelivery: True
+users:
+- githubUsername: myusername
+vendor:
+  label: redhat
+  name: Redhat
+```
+
+If these two conditions are met when the chart is submitted for certification, successful certification will not result in the chart being published in the OpenShift catalogue.
```
Comment (Srivaralakshmi) [inline docs/helm-chart-submission.md:59]: If these preceding conditions are met when the chart is submitted for certification, successful certification will not result in the chart being published in the OpenShift catalog.

**Review by Srivaralakshmi — COMMENTED** [review]
@mmulholla Please apply the changes suggested. Thanks!

File: docs/helm-chart-annotations.md [hunk 10]
```diff
@@ -1,7 +1,26 @@
...
+  - [charts.openshift.io/provider](#chartsopenshiftioprovider)
+  - [charts.openshift.io/name](#chartsopenshiftioname)
+  - [charts.openshift.io/supportURL](#chartsopenshiftiosupportURL)
+  - [charts.openshift.io/archs](#chartsopenshiftioarchs)  
+
+
 ## Verifier added annotations
 
+
 The chart-verifier tool adds annotations to a generated report, for example:
```
[the remark this answers is not in the document]
Comment (mmulholla) [reply docs/helm-chart-annotations.md:24]: Done.

File: docs/helm-chart-annotations.md [hunk 11]
```diff
@@ -18,6 +37,7 @@ metadata:
         lastCertifiedTimestamp: "2021-11-01T17:12:37.148895-04:00"
         testedOpenShiftVersion: 4.8
         supportedOpenShiftVersions: 4.5 - 4.8
+        providerControlledDelivery: false
```
[the remark this answers is not in the document]
Comment (mmulholla) [reply docs/helm-chart-annotations.md:40]: Done.

File: docs/helm-chart-annotations.md [hunk 12]
```diff
@@ -35,6 +55,7 @@ The annotations added differ based on the profiles version used:
 | [certifiedOpenShiftVersions](#certifiedOpenShiftVersions) | v1.0 
 | [testedOpenShiftVersion](#testedOpenShiftVersion)         | v1.1
 | [supportedOpenShiftVersions](#supportedOpenShiftVersions) | v1.1
+| [providerControlledDelivery](#providerControlledDelivery) | v1.0,v1.1
```
[the remark this answers is not in the document]
Comment (mmulholla) [reply docs/helm-chart-annotations.md:58]: Done.

[the remark this answers is not in the document]
Comment (mmulholla) [reply docs/helm-chart-annotations.md:58]: Done.

[the remark this answers is not in the document]
Comment (mmulholla) [reply docs/helm-chart-annotations.md:58]: Done

File: docs/helm-chart-submission.md (hunk 6 above)
Comment (mmulholla) [reply docs/helm-chart-submission.md:34]: Done.

File: docs/helm-chart-submission.md (hunk 7 above)
Comment (mmulholla) [reply docs/helm-chart-submission.md:43]: Done.

File: docs/helm-chart-submission.md (hunk 8 above)
Comment (mmulholla) [reply docs/helm-chart-submission.md:45]: This is actually a nice feature of md - label all in a numbered list 1. and it automatically creates correct numbering. Especially useful when adding a new list item in the middle of an existing numbered list

Comment (mmulholla) [reply docs/helm-chart-submission.md:45]: Done

File: docs/helm-chart-submission.md (hunk 9 above)
Comment (mmulholla) [reply docs/helm-chart-submission.md:59]: Done.

**Pull request merged**